src/PostgREST/Response.hs

```diff
@@ -68,6 +68,7 @@ actionResponse (DbCrudResult WrappedReadPlan{wrMedia, wrHdrsOnly=headersOnly, cr
     RSStandard{..} -> do
       let
         (status, contentRange) = RangeQuery.rangeStatusHeader iTopLevelRange rsQueryTotal rsTableTotal
+        cLHeader = if headersOnly then mempty else [contentLengthHeader rsBody]
```

Comment:
The body is not returned and processed in `HEAD` requests, so the length would be `0` here. The same resource using `GET` would return a length greater than `0`. [According to the spec](https://httpwg.org/specs/rfc9110.html#field.content-length) the `HEAD` and `GET` requests must return the same value in `Content-Length`, so we must not include the header altogether.

Paraphrased from:
> A server MAY send a Content-Length header field in a response to a HEAD request ([Section 9.3.2](https://httpwg.org/specs/rfc9110.html#HEAD)); a server MUST NOT send Content-Length in such a response unless its field value equals the decimal number of octets that would have been sent in the content of a response if the same request had used the GET method.